tests/global_scheduler/test_llm_engine_manager.py

```diff
@@ -0,0 +1,229 @@
+        self.instance_info = instance_info
+
+    def get_instance_info(self):
+        return self.instance_info
+
+    def is_ready(self) -> bool:
+        return True
+
+    def get_all_request_ids(self):
+        return list(self.request_id_set)
+
+    def get_num_request(self):
+        return self.num_request
+
+    def generate(self, request_id, server_info, *args, **kwargs):
+        self.request_id_set.add(request_id)
+        self.num_request = len(self.request_id_set)
+        return self.num_request
+
+    def abort(self, request_id):
+        if isinstance(request_id, str):
+            request_id = (request_id,)
+        request_ids = set(request_id)
+        for request_id in request_ids:
+            if request_id in self.request_id_set:
+                self.request_id_set.remove(request_id)
+                self.num_request = len(self.request_id_set)
+        return self.num_request
+
+    def migrate_out(self, dst_instance_name):
+        self.num_migrate_out += 1
+
+    def get_num_migrate_out(self):
+        return self.num_migrate_out
+
+
+def init_manager():
+    ray.init(ignore_reinit_error=True, namespace='llumnix')
+    try:
+        engine_manager_args = EngineManagerArgs()
+        engine_manager_args.log_instance_info = False
+        engine_manager = LLMEngineManager.from_args(engine_manager_args, None)
+    except ValueError:
+        engine_manager = ray.get_actor(MANAGER_ACTOR_NAME, namespace='llumnix')
+    ray.get(engine_manager.is_ready.remote())
+    return engine_manager
+
+def init_llumlets(initial_instances):
+    instance_ids = []
+    llumlets = []
+    for _ in range(initial_instances):
+        instance_id = random_uuid()
+        instance_name = 'instance_{}'.format(instance_id)
+        llumlet = MockLlumlet.options(name=instance_name,
+                                      namespace='llumnix').remote(instance_id)
+        instance_ids.append(instance_id)
+        llumlets.append(llumlet)
+    ray.get([llumlet.is_ready.remote() for llumlet in llumlets])
+    return instance_ids, llumlets
+
+@pytest.fixture
+def engine_manager():
+    engine_manager = init_manager()
+    ray.get(engine_manager.is_ready.remote())
+    yield engine_manager
+    ray.kill(engine_manager)
+    ray.shutdown()
+
+@pytest.fixture
+def llumlet():
+    instance_id = random_uuid()
+    instance_name = 'instance_{}'.format(instance_id)
+    llumlet = MockLlumlet.options(name=instance_name,
+                                  namespace='llumnix').remote(instance_id)
+    ray.get(llumlet.is_ready.remote())
+    return llumlet
+
+def test_init_manager(engine_manager):
+    assert engine_manager is not None
+    engine_manager_actor_handle = ray.get_actor(MANAGER_ACTOR_NAME, namespace='llumnix')
+    assert engine_manager_actor_handle is not None
+    assert engine_manager == engine_manager_actor_handle
+
+def test_init_llumlet(llumlet):
+    assert llumlet is not None
+    ray.get(llumlet.is_ready.remote())
```

Comment:
init_llumlets include llumlet initilization and scale up. The former can be tested by llumlet's unittest. The latter is already tested by manager. Because init_llumlets in manager will initialize actual instance which requires model and gpu, I do not test it temporarily and have added it to TODO. I plan to add this test after llumlet unittest development finished.